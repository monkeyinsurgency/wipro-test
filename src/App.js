import React, {Component} from 'react';
import Forecasts from './components/OWPForecast/OWPForecasts';

class App extends Component {
  render() {
    return (
      <Forecasts/>
    );
  }
}

export default App;
