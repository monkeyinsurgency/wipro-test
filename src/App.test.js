import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';
import testData from '../mocks/OWSMockData.json';

describe('<App />', () => {
  const data = testData;
  const config = { containerClassName: 'container-class-name' };

  const scenarios = [
    { description: 'w/o data w/o config', props: {} },
    { description: 'w data w/o config', props: { data } },
    { description: 'w/o data w config', props: { config } },
    { description: 'w data w config', props: { data, config } }
  ];

  scenarios.forEach((scenario) => {
    test(scenario.description, () => {
      const component = create(<App {...scenario.props} />);
      const tree = component.toJSON();
      console.log(tree);
      expect(tree).toMatchSnapshot();
    });
  });
});
