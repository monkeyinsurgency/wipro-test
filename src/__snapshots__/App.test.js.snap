// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> w data w config 1`] = `
<div
  className="forecast-view"
>
  <div
    className="city-name"
  >
    <span>
      5 Day Forecast for: 
      St Andrews, UK
    </span>
  </div>
  <div
    className="Loading"
  />
</div>
`;

exports[`<App /> w data w/o config 1`] = `
<div
  className="forecast-view"
>
  <div
    className="city-name"
  >
    <span>
      5 Day Forecast for: 
      St Andrews, UK
    </span>
  </div>
  <div
    className="Loading"
  />
</div>
`;

exports[`<App /> w/o data w config 1`] = `
<div
  className="forecast-view"
>
  <div
    className="city-name"
  >
    <span>
      5 Day Forecast for: 
      St Andrews, UK
    </span>
  </div>
  <div
    className="Loading"
  />
</div>
`;

exports[`<App /> w/o data w/o config 1`] = `
<div
  className="forecast-view"
>
  <div
    className="city-name"
  >
    <span>
      5 Day Forecast for: 
      St Andrews, UK
    </span>
  </div>
  <div
    className="Loading"
  />
</div>
`;
